### PR TITLE
Fix Redundant Computation of Marginal Likelihood for brmsfit Objects

### PR DESCRIPTION
**Pull request by ajnafa**
# Description

This PR adds a helper function to `bayesfactor_models.R` to check if `brmsfit` objects passed to `bayesfactor_*` functions already include a `marglik` element in the model object and avoids redundant calls to `bridgesampling::bridge_sampler` if the criteria already exists since this can be computationally expensive.

# Proposed Changes

- [x]  Add a helper function `.get_brmsfit_mML` to `bayesfactor_models.R`
- [x]  Modify `.bayesfactor_models_stan_REG` to check for `marglik` in `brmsfit` objects before calling `bridgesampling::bridge_sampler`
- [x]  Run unit tests to verify the changes didn't break anything else in the package

### CODE REVIEW

File: R/bayesfactor_models.R [hunk 1]
```diff
@@ -52,6 +52,8 @@
 #'   random effects) and their `log(BF)`s  (Use `as.numeric()` to extract the
 #'   non-log Bayes factors; see examples), that prints nicely.
 #'
+#' @importFrom stats median
```
Comment (strengejacke) [inline R/bayesfactor_models.R:55]: Please don't use `importFrom`, rather use `stats::median` directly below.

File: R/bayesfactor_models.R [hunk 2]
```diff
@@ -303,14 +305,22 @@ bayesfactor_models.default <- function(..., denominator = 1, verbose = TRUE) {
...
-    message("Computation of Bayes factors: estimating marginal likelihood, please wait...")
-  }
+  # Check the class of the models
+  check_brmsfit <- sapply(mods, class)
 
-  mML <- lapply(mods, function(x) {
-    bridgesampling::bridge_sampler(x, silent = TRUE)
-  })
+  # Add a check here for brmsfit object to avoid unnecessary computation of the ML
+  if (all(grepl("brmsfit", check_brmsfit))) {
```
Comment (strengejacke) [inline R/bayesfactor_models.R:312]: I suggest using`all(check_brmsfit == "brmsfit")`, which is faster:

``` r
check_brmsfit <- c("a", "b", "brmsfit", "c", "brmsfit")
microbenchmark::microbenchmark(
  all(grepl("brmsfit", check_brmsfit)),
  all(grepl("brmsfit", check_brmsfit, fixed = TRUE)),
  all(check_brmsfit == "brmsfit"),
  times = 1000
)
#> Unit: nanoseconds
#>                                                expr  min   lq   mean median
#>                all(grepl("brmsfit", check_brmsfit)) 5600 5800 6479.5   5900
#>  all(grepl("brmsfit", check_brmsfit, fixed = TRUE)) 1700 1900 2122.8   2000
#>                     all(check_brmsfit == "brmsfit")  300  400  440.6    400
```

**Review by strengejacke — CHANGES_REQUESTED** [review]

**Comment by mattansb** [comment]
@ajnafa LGTM

Can you add a unit test for this?

**Comment by ajnafa** [comment]
> @ajnafa LGTM
> 
> Can you add a unit test for this?

Yeah, I can add one to the test-bayesfactor_models.R file when I get into the office tomorrow.

**Comment by mattansb** [comment]
Awesome thanks!
(Sorry to interrupt your weekend - weekends here are Fri-Sat)

**Comment by ajnafa** [comment]
Wanted to follow up on this regarding unit tests because when I try to add the marginal likelihood to the brmsfit objects from within the package's project it crashes my R session. I'm currently trying to diagnose the issue but that is mostly the reason I haven't finished these yet.

**Comment by mattansb** [comment]
I just want to add another test of my own and then we can merge (:

**Comment by mattansb** [comment]
Looks good - should we address this in `NEWS`?

**Comment by strengejacke** [comment]
Isn't this PR ready to merge?

**Comment by bwiernik** [comment]
Looks okay ti me

**Comment by strengejacke** [comment]
Thanks, @ajnafa !

**Pull request merged**